### ISSUE TITLE
🚀 Feature: A python wrapper for the Novu API

### ISSUE THREAD

**Issue by ryshu** [novuhq/novu#2835]
### 🔖 Feature description

Implementation of a python wrapper to communicate quickly with the Novu API in Python projects

This is not really a feature request as we have already started the implementation, we mainly want to see with you the following issues:

* Do you want to transfer the project to your organization and take over the organization of the project?
* We were using the name "Novu" for the client for simplicity, but I think it's wrong. If you want to leave the repository isolated for the moment, we will probably have to see this point (change the name? or make an exemption under conditions?)
* See if you want to warn your python users that an unofficial client exists and that it may be interesting for them to use it

You can take a look at this project right here https://github.com/SpikeeLabs/novu-python. At the time I create this issue, the developments are still being validated by other developers in the alpha branch (and one of them has raised the licensing issue).

### 🎤 Why is this feature needed ?

As part of our Python project (and more specifically our Django microservices), we needed (SpikeeLabs) to implement a client to quickly communicate with your API.

The two main features of this client for us were:

* To be able to import a predefined notification template for our developers to store in each microservice (the microservice itself defines its notification flows)
* To be able to trigger these flows via a trigger.

In this context, I started the implementation of the latter based on:

* The API that you have defined via the openapi.yaml contract (https://api.novu.co/api/)
* Tests directly via Postman to check against reality (the contract has several errors)
* Typescript code reviews for the tricky cases and those I couldn't test myself (push notification for example)

Because I feel it can be useful to everyone, we put it in open source on Github, Pypi, and Readthedocs (not published because still in alpha).

### ✌️ How do you aim to achieve this?

The wrapper is built as follows:

* Definition of the objects that the API returns as DTO (https://en.wikipedia.org/wiki/Data_transfer_object)
* Definition of enumerations via python enumerations
* Definition of wrappers per resource, allowing interaction with the different endpoints of the API

### 🔄️ Additional Information

We also looked at the implementation you had started here https://github.com/novuhq/python/tree/initial-release, but you had several blocks on our side to exploit the latter:

* The project is in an abandoned state for 1 year
* We don't think it's your priority right now

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by ryshu** [comment]
I forgot to mention this, but we are open to the idea that anyone can edit/improve/fix the wrapper on the repository.

**Comment by scopsy** [comment]
We are working right now with to create a unified SDK for all languages for easier maintenance. Will update on the matter soon https://www.buildwithfern.com/ 🙏

**Comment by scopsy** [comment]
@Cliftonz could you please create an issue on Linear for the fern integration so we can track and link all relevant issues there?

**Comment by ryshu** [comment]
> We are working right now with to create a unified SDK for all languages for easier maintenance. Will update on the matter soon https://www.buildwithfern.com/ 🙏

Ok, so if I translate the answer you prefer that we release the name 'novu' in pypi?

**Comment by scopsy** [comment]
@ryshu yes, if you could keep the "novu" name available in pip until we decide if we can continue with fern officially so that we won't have to maintain a lot of SDK's and they will be generated automatically from our Open API Spec.

**Comment by ryshu** [comment]
Hi @scopsy,

We release the 'novu' name. In an internal discussion on the subject, we discussed the simplicity of taking the name and we recommend, to avoid problems, to reserve the name on PyPI (and on other dependency managers).

**Comment by ryshu** [comment]
FYI, it's now https://pypi.org/project/novu-python/ and we add a warning in README.md about this issue 
![image](https://user-images.githubusercontent.com/22155261/225451981-404b7108-9bf1-4101-88a8-bc8de2092fc9.png)

**Comment by unicodeveloper** [comment]
Hey @ryshu We have decided to **not continue** with the Fern SDK generation. Please can we revisit the convo about the SDK maintenance and making it official (by transferring it to Novu org) ? 

If you're on Discord, please let me know so that I can also ping you there as well. 🙏🏿

**Comment by ryshu** [comment]
Hey and thank @unicodeveloper

Sorry to hear that, it would have been a great solution to go quickly in the publication of SDKs for all languages.

I am at your disposal to move the repository and I can provide you with documentation for the configuration of the repository that I made (creation of the different accounts, tokens, etc...) so that you know how to take care of it 100%.

Same for the maintenance, available to discuss it (mainly during the week). I'll need to discuss this with my managers at SpikeeLabs to see how much time they will give me.

For the Discord part, I have an account but I rarely use it so I don't think it's very relevant.

**Comment by unicodeveloper** [comment]
Thanks @ryshu Really appreciate. Okay, if it's fine by you, we can move it during the week. What time during the week works for you?

**Comment by ryshu** [comment]
Hello @unicodeveloper ,

Sorry but my week got complicated. I'm usually available on Friday.

I still took the time this morning to prepare documentation for you to see the different actions I did to prepare the deposit. The other actions are automatic to simplify the management.

https://github.com/SpikeeLabs/novu-python/blob/main/SETUP.md

**Comment by unicodeveloper** [comment]
Hey @ryshu Thanks.

How's your week coming along. Went through the setup. 

So the easiest way to do the move is for you to transfer the repo to my account (@unicodeveloper). Once I get this, I'll transfer to Novu org. 

By default, I don't think this should affect the published package in PyPi.

Also, you will remain as a maintainer.

WDYT? @ryshu

**Comment by ryshu** [comment]
Much better than the previous one, thank you.

I made the transfer request to you now for the repo. 

For the Pypi, no immediate effect on the package. The best is for me to publish a new version by renaming the package "novu" and then we will remove the "novu-python" package.
You can make the new version when everything is ready on your side to test it.

**Comment by unicodeveloper** [comment]
Hey @ryshu, thanks for making the transfer. 

Now we have the package here: https://github.com/novuhq/novu-python 🚀  ❤️ 

1. Can you make the new version with the package name as just "novu"?
2. Do I need to create an account with pypi for novu to have the credentials?
3. When a new release is tagged on GitHub, does it automatically update the package release on PyPi?

**Comment by ryshu** [comment]
Hey @unicodeveloper,

I create an MR for that and some other useful things: https://github.com/novuhq/novu-python/pull/20

2. Yes, as I explain it [here](https://github.com/novuhq/novu-python/blob/main/SETUP.md#pypi), we need it for the `PYPI_TOKEN` variable
3. Yes, done by this workflow (and you can trigger a rebuild if we lose our package on PyPI): https://github.com/novuhq/novu-python/blob/main/.github/workflows/semver_build_publish.yaml

**Comment by ryshu** [comment]
Hey @scopsy  @unicodeveloper ,

The Python package is now available and handled by the NovuHQ org on Github and PyPi:

* https://github.com/novuhq/novu-python
* https://pypi.org/project/novu/

I will quickly cleanup the old one previously handled by SpikeeLabs https://pypi.org/project/novu-python/

**Comment by unicodeveloper** [comment]
Thank you so much @ryshu . Really appreciate the process!
